src/sentry/tasks/assemble.py

```diff
@@ -734,6 +745,23 @@ def _index_bundle_if_needed(self, release: str, dist: str, date_snapshot: dateti
                 metrics.incr("tasks.assemble.artifact_bundle.index_artifact_bundles_error")
                 sentry_sdk.capture_exception(e)
 
+    @sentry_sdk.tracing.trace
+    def _index_bundle_into_flat_file(self, artifact_bundle: ArtifactBundle):
+        identifiers = mark_bundle_for_flat_file_indexing(artifact_bundle)
```

Comment:
Any reason for not locking this part, we can save a bunch of queries in case of concurrency and we will not end up overwriting data for no reason, since concurrent execution of such function will result in the same update/creation.

Reply:
The problem is that such a lock would have to be very coarse-grained. And I doubt its worth it, as the records to Upsert here are very low. Certainly a lot lower than the `DebugIdArtifactBundle` we were inserting using the rather inefficient `create_or_update` previously.